source/blog/_posts/2018-08-23-against-microservices.md

```diff
@@ -0,0 +1,136 @@
+
+Problem solved? Well there are still the organizational difficulties...
+
+## Alternative: (Micro-)Kernel
+
+Microservices and monoliths are on oposing sides of the static dependency spectrum. While a monolith is interconnected and interdependent a Microservice is *(almost)* not. Well this spectrum does not only consist of black and white. My favorite alternative is the kernel style. A Kernel represents a core domain of an application. In our case eCommerce. This kernel then provides an extension mechanism for others to extend, alter and replace these core concepts. A plugin system if you want.
+
+Apart from the basic eCommerce workflows the Kernel provides the technical groundwork for all plugins to use. It contains a deployment machanism, package management and lays the technological groundwork for the necessarry infrastructure coupling. Most importantly it provides **the base quality of the product as a whole**.  
+
+In a company driven by Conways law the natural limit for a kernel is the amount of features a single team can handle. A basic runtime that shares a technical and functional base for everyone to harness. One arranges orthogonal features in orthogonal teams and creates a core that is owned and used by everyone. The viability of this approach is proven by big and small companies alike, think Operating System vendors, Framework vendors and the like. There is however a critical situation when a kernel's size outgrows the team. A need for action arises. A split is necessary.
+
+Conway's insight should mean that an organization in order to produce software needs to be fluid enough to support good design. So if an application changes the communication and therefore team structure needs to be changed in the best interest of the product.      
+
+## Conclusion
+
+From a technical standpoint there is no clear benefit in using Microservices as the main application design. They just have the potential to add risk and cost to a project. In my opinion this style just complicates stuff from the get go. Sizing is a really hard problem in software and I have seen many applications that where suprisingly large in relation to the work they actually performed. If you have a fairly simple problem to solve a single developer will be faster to fix it them twenty. Or as we say: 
+
+> Not everybody is Netflix
+
+If in the past you were not able to create a good application through concepts of OOP why would you be able to create such a thing adding layers of indirection on top of the actual problem. If the application design rots, there is no reason to expect adding networking to the communication will help. Microservices are by no means automatically cheaper in development or maintenance then a monolith. So when investing into new technology, maybe one should rather invest in solving the actuall problems of the past, directly.
```

Comment:
If you were not able 
OOP__,__ why
a thing __by__ adding layers

maintenance __than__ a monolith